docs/developer/extractor.md

```diff
@@ -0,0 +1,48 @@
+### Extractor
+
+Pumpkin maintains a FabricMC mod to generate the protocol information. This mod is written in Kotlin and uses the Gradle build toolchain to run FabricMC.
+
+It is found under the `extractor` folder on the repository.
+
+> [!IMPORTANT]
+> Pumpkin does not distribute any copyrightable information (game assets, programs), only information necessary to interoperate between clients (protocol information).
+
+### Generate
+
+1. Install the [Gradle](https://gradle.org/install/) build tool.
+
+2. Clone the git [repository](https://github.com/Snowiiii/Pumpkin) if you have not already.
+
+This will fail but will generate the `./extractor/run` folder:
+```
+cd extractor    
+gradle runServer
+```
+
+> [!NOTE]
+> All dependencies such as Minecraft server and FabricMC will be downloaded when running this command.
+
+
+3. Set `eula=false` in `extractor/run/eula.txt` to `eula=true` and then run this again:
+
+```
+gradle runServer
+```
+
+4. Wait until the last extractor completes to cancel the server
+
+5. The generated files will be under `./run/pumpkin_extractor_output`
+
+### Add an Extractor
+
+1. The simplest way to add a new extractor is to copy one (for example the `Particles.kt` file)
+2. Rename the file to your new registry
+3. Modify the file to add the required information according the internal structure of the Minecraft `Registry`
```

Comment:
You just need a File which implements the Extractor interface. Place it in the `extractors` folder and register it in the Main class. Done